get/install.bash

```diff
@@ -100,24 +137,53 @@ get_n_extract() {
         
 		;;
 	Linux)
-        STATIC_RELEASE_URL="https://github.com/tuzig/webexec/releases/download/v$LATEST_VERSION/webexec_${LATEST_VERSION}_$(uname -s | tr '[:upper:]' '[:lower:]')_$ARCH.tar.gz"
-       curl -sL "$STATIC_RELEASE_URL" | tar zx --strip-components=1
-       ./webexec init
+        BALL_NAME="webexec_${LATEST_VERSION}_$(uname -s | tr [:upper:] [:lower:])_$ARCH.tar.gz"
+        STATIC_RELEASE_URL="https://github.com/tuzig/webexec/releases/download/v$LATEST_VERSION/$BALL_NAME"
+        curl -sL "$STATIC_RELEASE_URL" -o $1/$BALL_NAME
+        tar zx --strip-components=1 -C $1 < $1/$BALL_NAME 
 	esac
 }
+
 do_install() {
 	init_vars
 	checks
 
-	tmp=$(mktemp -d)
+	sh_c='sh -c'
+	if [ "$user" != 'root' ]; then
+		if command_exists sudo; then
+			sh_c='sudo -E sh -c'
+		elif command_exists su; then
+			sh_c='su -c'
+		else
+			cat >&2 <<-'EOF'
+			Error: this installer needs the ability to run commands as root.
+			We are unable to find either "sudo" or "su" available to make this happen.
+			EOF
+			exit 1
+		fi
+	fi
+	lsb_dist=$( get_distribution )
+	lsb_dist="$(echo "$lsb_dist" | tr '[:upper:]' '[:lower:]')"
+	# Run setup for each distro accordingly
+	case "$lsb_dist" in
+		ubuntu|debian|raspbian)
+			pre_reqs="curl"
+            $sh_c 'apt-get update -qq >/dev/null'
+            $sh_c "DEBIAN_FRONTEND=noninteractive apt-get install -y -qq $pre_reqs >/dev/null"
+			;;
+    esac
 
-    echo "Created temp dir: $tmp"
-    cd $tmp
-    get_n_extract
+    tmp=$(mktemp -d)
+    echo "Created temp dir at $tmp"
+    get_n_extract $tmp
+    $tmp/webexec init
     # TODO: fixed launchd
-    if [[ "$(uname)" = Linux ]]; then
-        echo "==> We need su power to add webexec's binary and service"
-        sudo nohup bash replace_n_launch.sh $USER $HOME
+    if [ "$(uname)" = Linux ]; then
+        if debug; then
+            $sh_c "nohup bash replace_n_launch.sh $USER $HOME"
+        else
+            $sh_c "nohup bash $tmp/replace_n_launch.sh $USER $HOME"
```

Comment:
I assume `replace_n_launch.sh` now gets $tmp from $0 ? because you no longer `cd $tmp`.

Reply:
oops. You're right I should start the shell at $tmp. guess I always tested it from the projects' dir.